File: src/com/brianc/css/Color.java
package com.brianc.css;

public class Color {
	public final short r;
	public final short g;
	public final short b;
	public final short a;
	
	public static final Color BLACK = new Color(0, 0, 0, 255);

	public Color(short r, short g, short b, short a) {
		this.r = r;
		this.g = g;
		this.b = b;
		this.a = a;
	}
	
	public Color(int r, int g, int b, int a) {
		this((short)r, (short)g, (short)b, (short)a);
	}

	@Override
	public String toString() {
		return String.format("rgba(%d,%d,%d,%d)", r, g, b, a);
	}

}

File: src/com/brianc/css/Color.kt
package com.brianc.css

data class Color(val r: Short, val g: Short, val b: Short, val a: Short) {
    constructor(r: Int, g: Int, b: Int, a: Int) : this(r.toShort(), g.toShort(), b.toShort(), a.toShort())
    override fun toString() = "rgba($r,$g,$b,$a)"

    companion object {
        val BLACK = Color(0, 0, 0, 255)
    }
}
